Sources/TuistGenerator/Linter/TargetLinter.swift

```diff
@@ -61,9 +61,15 @@ class TargetLinter: TargetLinting {
     }
 
     private func lintProductName(target: Target) -> [LintingIssue] {
+        var productName = target.productName
+
+        // Remove any interpolated variables
+        productName = productName.replacingOccurrences(of: "\\$\\{.+\\}", with: "", options: .regularExpression)
+        productName = productName.replacingOccurrences(of: "\\$\\(.+\\)", with: "", options: .regularExpression)
+
         let allowed = CharacterSet(charactersIn: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_")
```

Comment:
What about including the `$`, `{` and `}` characters in this list instead?

Reply:
I thought about it, but as mentioned I followed what you did in #252 so thought you'd prefer it that way :P

Reply:
😂 there must be a reason for doing it that way but I can't remember it.